src/workerd/api/r2-bucket.h

```diff
@@ -256,14 +256,14 @@ class R2Bucket: public jsg::Object {
       return httpMetadata.map([](const HttpMetadata& m) { return m.clone(); });
     }
 
-    const jsg::Optional<jsg::Dict<kj::String>> getCustomMetadata() const {
-      return customMetadata.map([](const jsg::Dict<kj::String>& m) {
-        return jsg::Dict<kj::String>{
-          .fields =
-              KJ_MAP(f, m.fields) {
-          return jsg::Dict<kj::String>::Field{.name = kj::str(f.name), .value = kj::str(f.value)};
-        },
-        };
+    jsg::Optional<jsg::Dict<kj::String>> getCustomMetadata() {
+      return customMetadata.map([](const jsg::Dict<kj::String>& cm) {
+        auto res = jsg::Dict<kj::String>{};
+        res.fields.reserve(cm.fields.size());
+        for (auto& m: cm.fields) {
+          res.fields.insert(kj::str(kj::mv(m.key)), kj::str(kj::mv(m.value)));
```

Comment:
You're moving the `m.key` and `m.value` here, where the original was copying. If the `customMetadata` is not being consumed here then these should not move?